File: sources/extractor/internal/scip/testdata/fixtures/java-spring/src/main/java/example/CampaignRepository.java
package example;

/**
 * CampaignRepository is the data-access layer. The integration test
 * targets `deleteById` as the bottom of the call graph; the walker
 * must reach this from the controller via the service.
 */
public class CampaignRepository {

    public Campaign findById(String id) {
        // Stub implementation: production code would query a DB.
        return new Campaign(id);
    }

    public void deleteById(String id) {
        // Stub: production code would delete from a DB.
    }
}

File: internal/extractor/scip/testdata/fixtures/kotlin-spring/src/main/kotlin/example/CampaignRepository.kt
package example

/**
 * CampaignRepository is the data-access layer. The integration test
 * targets `deleteById` as the bottom of the call graph; the walker
 * must reach this from the controller via the service.
 */
class CampaignRepository {
    fun findById(id: String): Campaign? = Campaign(id)
    fun deleteById(id: String) {
        // Stub: production code would delete from a DB.
    }
}
